src/components/sidenav/sidenav.js

```diff
@@ -278,31 +307,51 @@ function SidenavDirective($mdMedia, $mdUtil, $mdConstant, $mdTheming, $animate,
      * Toggle the SideNav view and attach/detach listeners
      * @param isOpen
      */
-    function updateIsOpen(isOpen) {
+    function updateIsOpen(isOpen, oldValue) {
+      if (skipNextUpdate) {
+        skipNextUpdate = false;
+        return;
+      }
+
       // Support deprecated md-sidenav-focus attribute as fallback
       var focusEl = $mdUtil.findFocusTarget(element) || $mdUtil.findFocusTarget(element,'[md-sidenav-focus]') || element;
       var parent = element.parent();
 
+      // Temporary remove md-closed class (won't affect the view, because it's outside of the viewport).
+      // And then check the visbility due hide attributes. So if the isOpen variable changes
+      // and a hide attribute is active we should revert the digest change
+      if (!skipSidenav) {
+        var wasClosed = element.hasClass('md-closed');
+        element.removeClass('md-closed');
+        if ($mdUtil.isHidden(element, true, false) && !skipSidenav) {
+          element.toggleClass('md-closed', wasClosed);
+          skipNextUpdate = true;
+          scope.isOpen = !!oldValue;
+          return;
+        }
+        element.toggleClass('md-closed', wasClosed);
+      }
+
       parent[isOpen ? 'on' : 'off']('keydown', onKeyDown);
       backdrop[isOpen ? 'on' : 'off']('click', close);
 
-      if ( isOpen ) {
+      if ( isOpen && !skipSidenav) {
         // Capture upon opening..
         triggeringElement = $document[0].activeElement;
       }
 
       disableParentScroll(isOpen);
 
-      return promise = $q.all([
-                isOpen ? $animate.enter(backdrop, parent) : $animate.leave(backdrop),
-                $animate[isOpen ? 'removeClass' : 'addClass'](element, 'md-closed')
-              ])
-              .then(function() {
-                // Perform focus when animations are ALL done...
-                if (scope.isOpen) {
-                  focusEl && focusEl.focus();
-                }
-              });
+      var actions = [isOpen ? $animate.enter(backdrop, parent) : $animate.leave(backdrop)];
+      if (!skipSidenav) actions.push(isOpen ? $animate.removeClass(element, 'md-closed') : $animate.addClass(element, 'md-closed'));
```

Comment:
Done :hand: I should definitely use that `conventions` (how I call it) in my next PR's so we would save much time :smile: - Already done in my newer PR's